bgpd/bgp_vty.c

```diff
@@ -8786,33 +9292,461 @@ static void bgp_show_peer_afi_orf_cap(struct vty *vty, struct peer *p,
+		if (bgp_global_gr_mode_get(p->bgp) == GLOBAL_HELPER)
+			mode = "Helper*";
+		else if (bgp_global_gr_mode_get(p->bgp) == GLOBAL_GR)
+			mode = "Restart*";
+		else if (bgp_global_gr_mode_get(p->bgp) == GLOBAL_DISABLE)
+			mode = "Disable*";
+		else
+			mode = "Invalid*";
+	else
+		mode = "Invalid";
+
+	if (use_json) {
+		json_object_string_add(json,
+				"localGrMode", mode);
+	} else {
+		vty_out(vty, mode, "\n");
+	}
+}
+
+static void  bgp_show_neighbor_graceful_restart_capability_per_afi_safi(
+					struct vty *vty, struct peer *peer,
+					bool use_json, json_object *json)
+{
+	afi_t afi = AFI_MAX;
+	safi_t safi = SAFI_MAX;
+	json_object *json_afi_safi = NULL;
+	json_object *json_timer = NULL;
+	json_object *json_endofrib_status = NULL;
+	bool eor_flag = false;
+
+	for (afi = AFI_IP; afi < AFI_MAX; afi++) {
+		for (safi = SAFI_UNICAST; safi <= SAFI_MPLS_VPN; safi++) {
+			if (peer->afc[afi][safi]
+				&& CHECK_FLAG(peer->cap, PEER_CAP_RESTART_ADV)
+				&& CHECK_FLAG(peer->cap, PEER_CAP_RESTART_RCV)
+				) {
+				if (use_json) {
+					json_afi_safi =
+						json_object_new_object();
+					json_endofrib_status =
+						json_object_new_object();
+					json_timer =
+						json_object_new_object();
+				}
+
+				if (peer->eor_stime[afi][safi] >=
+				    peer->pkt_stime[afi][safi])
+					eor_flag = true;
+				else
+					eor_flag = false;
+
+				if (!use_json) {
+					vty_out(vty, "    %s :\n",
+					get_afi_safi_str(afi, safi, false));
+
+					vty_out(vty,
+					"     F bit                 : ");
+				} else
+					get_afi_safi_str(afi, safi, true);
+
+				if (peer->nsf[afi][safi]
+					&& CHECK_FLAG(
+					peer->af_cap[afi][safi],
+					PEER_CAP_RESTART_AF_PRESERVE_RCV)) {
+
+					if (use_json) {
+						json_object_boolean_true_add(
+						json_afi_safi, "fBit");
+					} else {
+						vty_out(vty,
+							"True\n");
+					}
+
+				} else {
+
+					if (use_json) {
+						json_object_boolean_false_add(
+						json_afi_safi, "fBit");
+					} else {
+						vty_out(vty,
+							"False\n");
+					}
+
+				}
+
+				if (!use_json)
+					vty_out(vty,
+					"     End-of-RIB Received   : ");
+
+				if (CHECK_FLAG(peer->af_sflags[afi][safi],
+						PEER_STATUS_EOR_RECEIVED)) {
+
+					if (use_json) {
+						json_object_boolean_true_add(
+						json_endofrib_status,
+							"endOfRibRecv");
+					} else {
+						vty_out(vty, "Yes\n");
+					}
+
+				} else {
+					if (use_json) {
+						json_object_boolean_false_add(
+						json_endofrib_status,
+							"endOfRibRecv");
+					} else {
+						vty_out(vty, "No\n");
+					}
+				}
+
+				if (!use_json)
+					vty_out(vty,
+					"     End-of-RIB Send       : ");
+
+				if (CHECK_FLAG(peer->af_sflags[afi][safi],
+						PEER_STATUS_EOR_SEND)) {
+					if (use_json) {
+						json_object_boolean_true_add(
+						json_endofrib_status,
+							 "endOfRibSend");
+
+						PRINT_EOR_JSON(eor_flag);
+					} else {
+						vty_out(vty, "Yes\n");
+						vty_out(vty,
+						"     EoRSentAfterUpdate   : ");
```

Comment:
since this is not a json, can we have a human-readable format?

Reply:
Actually if i change the string now, the o/p alignment will go for a toss. So leaving is as it is